src/wh_nvm_flash.c

```diff
@@ -578,6 +621,13 @@ static int nfObject_ReadDataBytes(whNvmFlashContext* context, int partition,
     start = context->directory.objects[object_index].state.start;
     startOffset = nfPartition_DataOffset(context, partition) + start;
 
+    /* Ensure we don't read off the end of the active partition */
+    if (WH_ERROR_OK != nfPartition_CheckRangeUnits(context, partition,
+                                    startOffset + WHFU_BYTES2UNITS(byte_offset),
+                                    WHFU_BYTES2UNITS(byte_count))) {
```

Comment:
Fix the math bug (BYTES2UNITS always rounds up) by making this byte based:
```suggestion
    if (WH_ERROR_OK != nfPartition_CheckDataRange(context, partition,
                                    startOffset * WHFU_BYTESPERUNIT + byte_offset,
                                   byte_count)) {
```